src/hooks/useSingleLineData.ts

```diff
@@ -35,17 +37,45 @@ export const useSingleLineData = (lineRef?: number, routeIds?: number[]) => {
     return pos
   }, [locations])
 
+  function convertTo24HourAndToNumber(time: string): number {
```

Comment:
The name does not reflect that the output of this method is the number of minutes since midnight. Would you consider renaming this method?
This is my suggestion, but feel free to implement anything you like.
```suggestion
  function timestringToMinutes(time: string): number {
```